src/api/spec/models/project_spec.rb

```diff
@@ -453,4 +452,20 @@
       end
     end
   end
+
+  describe "#destroy" do
+    context 'avoid regressions of the issue #3665' do
+      let(:admin_user) { create(:admin_user, login: 'Admin') }
+      let(:images_repository) { create(:repository, name: "images", project: project) }
+      let(:apache_repository) { create(:repository, name: "Apache", project: project) }
+      let!(:path_element) { create(:path_element, parent_id: images_repository.id, repository_id: apache_repository.id, position: 1)}
+
+      before do
+        login admin_user
+        project.destroy!
+      end
+
+      it { expect(Project.deleted?(project.name)).to be_truthy }
```

Comment:
No, with this setup...
In this case with having 2 repos, one linking to the other in the same project, the delete operation is executed in the backend and right after that a put with the meta was performed, so the backend had a project already after deleting it.

That's the cause of the issue, this test is only for avoiding the regression.